packages/create-commandkit/src/functions/setup.ts

```diff
@@ -0,0 +1,37 @@
+import type { ModuleType, PackageManager } from '../types';
+import { type IOType, execSync } from 'child_process';
+import { commands } from '../utils';
+
+import fs from 'fs-extra';
+import path from 'path';
+
+interface SetupProps {
+    manager: PackageManager;
+    type: ModuleType;
+    token: string;
+    dir: string;
+    stdio?: IOType;
+}
+
+export async function setup({ manager, type, token, dir, stdio = 'pipe' }: SetupProps) {
+    await fs.emptyDir(dir);
+    execSync(commands.init[manager], { cwd: dir, stdio });
+
+    const packageJsonPath = path.join(dir, 'package.json');
+    const packageJson = await fs.readJSON(packageJsonPath);
+
+    packageJson.name = packageJson.name.toLowerCase();
+    packageJson.type = type == 'esm' ? 'module' : 'commonjs';
+
+    packageJson.main = './src/index.js';
```

Comment:
Main entry point could be misleading with cli below.

Reply:
> Main entry point could be misleading with cli below.

how is this misleading? It's the same entry point — you can use either `node .` or the npm scripts. When TypeScript templates are added, this will be changed to the build output.

Reply:
`commandkit dev` does not execute from this path. Add a note maybe, indicating that `main` is only applicable outside of commandkit cli.